Arm/Memory/AddressNormalization.lean

```diff
@@ -0,0 +1,267 @@
+/-
+Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Siddharth Bhat, Tobias Grosser
+-/
+
+/-
+This file implements bitvector expression simplification simprocs.
+We perform the following additional changes:
+
+1. Canonicalizing bitvector expression to always have constants on the left.
+  Recall that the default associativity of addition is to the left: x + y + z = (x + y) + z.
+  If we thus normalize our expressions to have constants on the left,
+  and if we constFoldiate our additions to be to the left, we will naturally perform
```

Comment:
```suggestion
  and if we constant-fold our additions to be to the left, we will naturally perform
```

Reply:
This sentence still feels a bit broken, but either way, please rephrase constFoldiate